### PR TITLE
Fix masking add composability

### PR DESCRIPTION
**Pull request by dPys**
@oesteban -- It is truly incredible how far this repo has come. I also still believe it remains right on the cusp of making eddy correction of dmri more accessible to researchers looking for other alternatives to FSL. 

There's very little to complain about with how you've organized things already. With that said, I do have some proposed enhancements, any or none of which you're welcome to adopt. I realize that this first PR comes after a long lull in my contribution, so it mainly just includes some general housekeeping, most of which comes from debugging locally with test data that includes a b0 mask. Here is a brief summary of the included changes:

* Relegates registration logic previously belonging to the `EddyMotionEstimator` class to a more modular Aligner dataclass. This may ultimately also make it easier to surgically swap out nipype-based native ANTs with ANTspy, if that strategy is still on the agenda per #89 and #90 .
* Refactors the order and methodology of parsing `S0` and `mask`  from kwargs to address cases where either or both are manually provided by the user. In the case of a B0 mask, for example, I found that np.ma.masked_array behaves more predictably than simple slicing. These use-cases probably also need to further unit tests before this PR gets merged. Per #107 , it might also be worth proceeding to raise a warning if a mask is not provided.
* Adds some additional runtime error handling and simplified results concatenation logic around the joblib calls in the fit/predict logic of `BaseModel`.
* Collectively, these improvements should help to shave off some memory consumption and lead to a small speedup overall.
* My linter auto-reformatted some things per black as well as flake8, so feel free to reject those changes if you have a different formatting aesthetic in mind.

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -2,8 +2,8 @@ Anisha Keshavan <akeshavan@gmail.com> <akeshavan@ucla.edu>
 Anisha Keshavan <akeshavan@gmail.com> <keshavan@berkeley.edu>
 Ariel Rokem <arokem@berkeley.edu>
 Ariel Rokem <arokem@berkeley.edu> <arokem@gmail.com>
-Derek Pisner <dpisner@utexas.edu>
-Derek Pisner <dpisner@utexas.edu> dPys <dpisner@utexas.edu>
+Derek Pisner <dpysalexander@gmail.com>
+Derek Pisner <dpysalexander@gmail.com> dPys <dpysalexander@gmail.com>
```
Comment (oesteban) [inline .mailmap:6]: This will remove your name from old commits done with the old email.

That said, please send a separate PR with this edit.
```suggestion
Derek Pisner <dpysalexander@gmail.com> dPys <dpysalexander@gmail.com>
Derek Pisner <dpysalexander@gmail.com> <dpisner@utexas.edu> 
```

File: setup.cfg [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [metadata]
 name = eddymotion
 author = Derek Pisner
-author_email = dpisner@utexas.edu
+author_email = dpysalexander@gmail.com
```
Comment (oesteban) [inline setup.cfg:4]: Like the other, please pull out on the other PR. I think we should also follow the general nipreps guidelines and write the nipreps account.

```suggestion
author_email = nipreps@gmail.com
```

File: setup.cfg [hunk 3]
```diff
@@ -15,7 +15,7 @@ license = Apache License, Version 2.0
 long_description = file:README.rst
 long_description_content_type = text/x-rst; charset=UTF-8
 maintainers = The NiPreps developers
-maintainer_email = nipreps@gmail.com, dpisner@utexas.edu
+maintainer_email = nipreps@gmail.com, dpysalexander@gmail.com
```
Comment (oesteban) [inline setup.cfg:18]: Ditto.

**Review by oesteban — COMMENTED** [review]
I need to revise the PR, but for the time being, please separate the packaging/metadata changes.

File: .mailmap (hunk 1 above)
Comment (dPys) [reply .mailmap]: See https://github.com/nipreps/eddymotion/pull/119

File: setup.cfg (hunk 2 above)
Comment (dPys) [reply setup.cfg]: Ditto, see https://github.com/nipreps/eddymotion/pull/119

File: setup.cfg (hunk 3 above)
Comment (dPys) [reply setup.cfg]: see https://github.com/nipreps/eddymotion/pull/119

**Comment by dPys** [comment]
No rush, re: any revisions.

Not sure how my contact info updates slipped into my final commit-- sorry about that!  RE: packaging/metadata changes, I do actually have some more related to `pyproject.toml` specifically, but will scope out those commits to a third PR once this one and https://github.com/nipreps/eddymotion/pull/119 are over the finish-line.

**Comment by oesteban** [comment]
We have worked several aspects of this PR elsewhere, so I think it's safe to close. Thanks to @dPys for the effort - we've learned a lot from this.
